src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/PackageAnnotationCheck.java

```diff
@@ -30,19 +30,20 @@
  * Checks that all package annotations are in the package-info.java file.
  * </p>
  * <p>
- * According to the Java Language Specification.
- * </p>
- * <p>
- * The JLS does not enforce the placement of package annotations.
- * This placement may vary based on implementation. The JLS
- * does highly recommend that all package annotations are
- * placed in the package-info.java file.
+ * According to the Java Language Specification, it is highly recommended
+ * that all package annotations are placed in the package-info.java file.
+ * However, this placement is not enforced for all implementations.
  *
- * See <a
- * href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-7.html#jls-7.4.1">
+ * See <a href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-7.html#jls-7.4.1">
  * Java Language Specification, &#167;7.4.1</a>.
  * </p>
  * <p>
+ * More precisely, for SE8 and above,
```

Comment:
I wonder about leading with this - what do you think? Maybe something like

>Checks that all package annotations are in the package-info.java file.

> For Java SE8 and above, placement of package annotations in the package-info.java file is enforced by the compiler and this check is not necessary.

>For Java SE7 and below, the Java Language Specification highly recommends but doesn't require that annotations are placed in the package-info.java file, and this check can help enforce that placement.

Reply:
I think it's much clearer, thanks.
PR is updated.